### PR TITLE
[CPDLP-3519] Add contract for course validation to change schedule and declarations create services

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

Ticket: [CPDLP-3519](https://dfedigital.atlassian.net/browse/CPDLP-3519)

We want to add current contract logic in record declaration and change schedules.

### Changes proposed in this pull request

Add contract for course validation to change schedule and declarations create services.

[CPDLP-3519]: https://dfedigital.atlassian.net/browse/CPDLP-3519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/services/participants/change_schedule.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class ChangeSchedule < Action
 
     validates :schedule_identifier, presence: true
     validate :validate_new_schedule_found
-    # validates :cohort, npq_contract_for_cohort_and_course: true # TODO we don't have NPQ Contract yet
+    validates :cohort, contract_for_cohort_and_course: true
```
Comment (cwrw) [inline app/services/participants/change_schedule.rb:11]: having a think about this, do we really need it here, in ECF it's for funding cap, here we get the cap from cohort so maybe we don't need to check the contract after all?

File: spec/services/participants/change_schedule_spec.rb [hunk 2]
```diff
@@ -56,14 +58,9 @@
...
-        before do
-          create(:schedule, :npq_leadership_autumn, cohort:)
-        end
+        let(:new_schedule) { create(:schedule, :npq_leadership_spring, cohort:) }
 
-        it "is invalid and returns an error message" do
-          expect(subject.change_schedule).to be_truthy
-          expect(subject).to have_error(:schedule_identifier, :already_on_the_profile, "Selected schedule is already on the profile")
-        end
+        it { is_expected.to have_error(:schedule_identifier, :already_on_the_profile, "Selected schedule is already on the profile") }
```
Comment (cwrw) [inline spec/services/participants/change_schedule_spec.rb:63]: any reason we changed this test? as it was testing attempting the change twice? we could set up a new contract for that course?

**Review by cwrw — COMMENTED** [review]
looks good, just a minor question :)

File: spec/services/participants/change_schedule_spec.rb (hunk 2 above)
Comment (leandroalemao) [reply spec/services/participants/change_schedule_spec.rb:63]: This spec was failing after the changes but I realised it was kind confusing as well with the first expect to be truthy and the second expecting an error:

`expect(subject.change_schedule).to be_truthy`
`expect(subject).to have_error(:schedule_identifier, :already_on_the_profile, "Selected schedule is already on the profile")`

Make the changes to keep it in line with the test above as well.

**Review by cwrw — APPROVED** [review]

**Review by mooktakim — APPROVED** [review]

**Pull request merged**